src/AzureAuth/CommandMain.cs

```diff
@@ -394,35 +396,36 @@ private int GetToken()
 
                     try
                     {
-                        result = authFlow.GetTokenAsync().Result;
+                        results = authFlowExecutor.GetTokenAsync().Result;
+                        succeededResult = results.FirstOrDefault(result => result.Success == true);
                     }
                     finally
                     {
                         mutex.ReleaseMutex();
                     }
                 }
 
-                // But what if result is null? The compiler cannot ensure it won't be (yet).
-                this.eventData.Add("error_list", ExceptionListToStringConverter.Execute(result.Errors));
+                var errors = results.SelectMany(result => result.Errors).ToList();
+                this.eventData.Add("error_list", ExceptionListToStringConverter.Execute(errors));
 
-                if (!result.Success)
+                if (succeededResult == null)
                 {
                     this.logger.LogError("Authentication failed. Re-run with '--verbosity debug' to get see more info.");
                     return 1;
                 }
 
-                this.eventData.Add("auth_type", $"{result.TokenResult.AuthType}");
+                this.eventData.Add("auth_type", $"{succeededResult.TokenResult.AuthType}");

```

Comment:
By this point you know you've got a non-null `succeededResult` and that it has a `TokenResult` on it. You could make the lines below a bit shorter if you pulled it out. For example
```suggestion
                var tokenResult = succeededResult.TokenResult;
                this.eventData.Add("auth_type", $"{tokenResult.AuthType}");

```
Doesn't save you much for this one line, but it does make this one _and_ the following 3 a little cleaner.